tests/test_template_types.py

```diff
@@ -928,43 +928,41 @@ def test_letter_output_pipe_delimiting():
             'address line 6': '',
             'postcode': 'NR1 5PQ',
         },
-        numeric_id=1,
+        notification_reference="1",
     )
 
     assert len(str(template).split('|')) == len(dvla_file_spec)
     assert 'Pipes  pipes  everywhere' in str(template)
 
 
 @pytest.mark.parametrize('id, expected_exception', [
-    (None, 'numeric_id is required'),
-    ('abc123', 'numeric_id must be an integer'),
-    (12345678, 'numeric_id cannot be longer than 7 digits'),
-    (1.34567, 'numeric_id must be an integer')
+    (None, 'notification_reference is required'),
+    ("01234567891234567", 'notification_reference cannot be longer than 16 chars')
 ])
 def test_letter_output_numeric_id(id, expected_exception):
     with pytest.raises(TypeError) as error:
         LetterDVLATemplate(
             {'content': '', 'subject': ''},
-            numeric_id=id,
+            notification_reference=id,
         )
     assert str(error.value) == expected_exception
 
 
-def test_letter_output_stores_valid_numeric_id():
+def test_letter_output_stores_valid_notification_reference():
     assert LetterDVLATemplate(
         {'content': '', 'subject': ''},
-        numeric_id=1234567,
-    )._numeric_id == 1234567
+        notification_reference="1234567",
+    ).notification_reference == "1234567"
 
 
 @pytest.mark.parametrize('extra_args, expected_field', [
     ({}, '500'),
     ({'org_id': '001'}, '001'),
 ])
-def test_letter_output_numeric_id(extra_args, expected_field):
+def test_letter_output_notification_reference(extra_args, expected_field):
```

Comment:
This test should be called `test_letter_outputs_org_id` or something similar. My bad.